### ISSUE TITLE
Rule S4798: Boolean parameters should not be optional

### ISSUE THREAD

**Issue by saberduck** [SonarSource/SonarTS#544]
if possible prefer providing default value rather than making boolean parameter optional

```
function foo(arg?: boolean) // Noncompliant
function foo(arg = true) // Compliant
```

**Comment by vilchik-elena** [comment]
https://jira.sonarsource.com/browse/RSPEC-4798

**Comment by schlundus** [comment]
I think there are some inconsistencies with the ts syntax while checking the rule

Please see the following example, both functions are raising an "Optional boolean parameters should have default value" issue. 
`
function foo(a: number, b?: number);
function foo(a: number, b?: boolean);
function foo(a: number, b?: any): void {
   
}

abstract class Test {
    public abstract test(a: number, b?: boolean);
}
`

But correcting them to....  the following raises an error on TS (see playground)

`
function foo(a: number, b?: number);
function foo(a: number, b = true);
function foo(a: number, b?: any): void {
   
}

abstract class Test {
    public abstract test(a: number, b = true);
}
`
Ay hint regarding this one? Bug?

**Comment by andrea-guarino-sonarsource** [comment]
hi @schlundus,

Thank you for reporting this.
This is indeed a false positive: I opened #823 to keep track of this issue.
